File: benchmark/perl/matrix.pl
use Time::HiRes qw(time);
my $size = 150;
my (@a, @b, @c);
for my $i (0..$size-1) {
    for my $j (0..$size-1) {
        $a[$i][$j] = $i + $j;
        $b[$i][$j] = $i - $j;
        $c[$i][$j] = 0;
    }
}
my $start = time();
for my $i (0..$size-1) {
    for my $j (0..$size-1) {
        my $s = 0;
        for my $k (0..$size-1) { $s += $a[$i][$k] * $b[$k][$j]; }
        $c[$i][$j] = $s;
    }
}
my $end = time();
my $ms = ($end - $start) * 1000;
print "Result[0][0]: $c[0][0]
";
print "Result[149][149]: $c[149][149]
";
printf "Time: %.0fms
", $ms;

File: benchmark/lua/matrix.lua
local size = 150
local a = {}
local b = {}
local c = {}
for i = 1, size do
    a[i] = {}
    b[i] = {}
    c[i] = {}
    for j = 1, size do
        a[i][j] = (i - 1) + (j - 1)
        b[i][j] = (i - 1) - (j - 1)
        c[i][j] = 0
    end
end

local start = os.clock()
for i = 1, size do
    for j = 1, size do
        for k = 1, size do
            c[i][j] = c[i][j] + a[i][k] * b[k][j]
        end
    end
end
local finish = os.clock()
print("Result[0][0]: " .. c[1][1])
print("Result[149][149]: " .. c[150][150])
print(string.format("Time: %.0fms", (finish - start) * 1000))

